Patch 1

cjhatch opened the pull request:
This provides updated instructions for the Windows install

cjhatch committed:
Update README.md

Clarified install for Windows users to install the HughesLabFolder and not the HughesLabFolder-main

cjhatch committed:
Update README.md

aforsythe commented:
Should the install be a manual process on windows?
Why not use the powershell script?

Here's a shot at a powershell script with similar features to the shell script.
I can't test but let me know if this is useful.

```
Param(
    [switch]$Verbose
)

$ErrorActionPreference = 'Stop'

function Debug-Print($msg) {
    if ($Verbose) { Write-Host $msg }
}

function Test-FijiValid($path) {
    Debug-Print "Validating Fiji at path: $path"
    # Adjust executable name if your Fiji uses a different filename
    $exe = Join-Path $path 'ImageJ-win64.exe'
    if (Test-Path (Join-Path $path 'jars') -PathType Container `
        -and Test-Path (Join-Path $path 'plugins') -PathType Container `
        -and Test-Path $exe -PathType Leaf) {
        Debug-Print "Fiji validation passed."
        return $true
    } else {
        Debug-Print "Fiji validation failed."
        return $false
    }
}

function Install-Fiji {
    if (-not (Get-Command winget -ErrorAction SilentlyContinue)) {
        $choice = Read-Host "winget not found. Please install the Windows Package Manager App Installer from the Microsoft Store, then re-run"
        exit 1
    }
    Write-Host "Installing Fiji via winget..."
    winget install --id Fiji.Fiji -e --accept-package-agreements --accept-source-agreements
}

function Prompt-InstallFiji {
    $choice = Read-Host "Would you like to install Fiji now? (y/n)"
    if ($choice -match '^[Yy]') { Install-Fiji }
    else { Write-Host "Fiji is required to proceed, exiting."; exit 1 }
}

#
# 1) Check or install Fiji
#
$FIJIDir = 'C:\Program Files\Fiji'
if (-not (Test-FijiValid $FIJIDir)) {
    Write-Host "No valid Fiji found in $FIJIDir."
    $alt = Read-Host "Specify alternative path? (y/n)"
    if ($alt -match '^[Yy]') {
        $path = Read-Host 'Enter full Fiji install path (e.g. C:\Program Files\Fiji)'
        if (-not (Test-Path $path)) {
            Write-Host "Path '$path' does not exist."
            Prompt-InstallFiji
        }
        if (Test-FijiValid $path) {
            $FIJIDir = $path
            Write-Host "Using Fiji at $FIJIDir."
        } else {
            Write-Host "Not a valid Fiji installation."
            Prompt-InstallFiji
        }
    } else {
        Prompt-InstallFiji
    }
} else {
    Write-Host "Fiji detected in $FIJIDir."
}

#
# 2) Prepare directories
#
$libDir     = Join-Path $FIJIDir 'jars\Lib'
$scriptsDir = Join-Path $FIJIDir 'scripts\Hughes_Lab_Tools'
New-Item -Path $libDir     -ItemType Directory -Force | Out-Null
New-Item -Path $scriptsDir -ItemType Directory -Force | Out-Null

#
# 3) Locate HughesLabTools in the repo
#
$ScriptDir   = Split-Path -Parent $MyInvocation.MyCommand.Definition
$HughesDir   = Join-Path $ScriptDir 'src\HughesLabTools'
if (-not (Test-Path $HughesDir)) {
    Write-Host "Error: '$HughesDir' not found."; exit 1
}
$MainScript  = Join-Path $HughesDir 'main_.py'
if (-not (Test-Path $MainScript)) {
    Write-Warning "main_.py missing, menu item may not appear."
}

#
# 4) Prompt for install type
#
$options = @(
    'Copy Hughes Lab Tools (End-user Mode)',
    'SymLink Hughes Lab Tools (Developer Mode)',
    'Uninstall Hughes Lab Tools',
    'Quit'
)
for ($i=0; $i -lt $options.Count; $i++) {
    Write-Host "[$($i+1)] $($options[$i])"
}
$sel = Read-Host "Choose an option [1-$($options.Count)]"

switch ($sel) {
    '1' {
        Write-Host "Copying HughesLabTools..."
        $dest = Join-Path $libDir 'HughesLabTools'
        Remove-Item -Recurse -Force $dest -ErrorAction SilentlyContinue
        Debug-Print "Copying $HughesDir → $dest"
        Copy-Item $HughesDir -Destination $dest -Recurse -Force -Verbose:$Verbose

        if (Test-Path $MainScript) {
            $link = Join-Path $scriptsDir 'All_VMO_Tools.py'
            New-Item -Path $link -ItemType SymbolicLink `
                -Value (Join-Path $dest 'main_.py') -Force -Verbose:$Verbose
        }
    }
    '2' {
        Write-Host "Creating symbolic links..."
        $dest = Join-Path $libDir 'HughesLabTools'
        Remove-Item -Recurse -Force $dest -ErrorAction SilentlyContinue
        Debug-Print "Linking $HughesDir → $dest"
        New-Item -Path $dest -ItemType SymbolicLink `
            -Value $HughesDir -Force -Verbose:$Verbose

        if (Test-Path $MainScript) {
            $link = Join-Path $scriptsDir 'All_VMO_Tools.py'
            New-Item -Path $link -ItemType SymbolicLink `
                -Value $MainScript -Force -Verbose:$Verbose
        }
    }
    '3' {
        Write-Host "Uninstalling HughesLabTools..."
        Remove-Item -Recurse -Force (Join-Path $libDir 'HughesLabTools'), $scriptsDir -ErrorAction SilentlyContinue
    }
    '4' {
        Write-Host "No changes were made; exiting."
    }
    default {
        Write-Host "Invalid choice; exiting."
    }
}

Write-Host "Done."
```

cjhatch committed:
Updated DeviceImage.py

- Changed instructions for cropping to instruct user to draw a rectangle instead of adjusting it

cjhatch committed:
Updated DeviceManager.py

- Allows for png images to be used
- updated the directories to skip the Vessel_Thresholded and Tumor_Segmented
- Changed naming for Weka folder from Tumor_Segmented to Tumor_Segmented_Weka, and changed non weka folder from segmented to Tumor_Segmented

cjhatch committed:
Updated TumorImage.py and VesselsImage.py

- Changed naming for Weka folder from Tumor_Segmented to Tumor_Segmented_Weka, and changed non weka folder from segmented to Tumor_Segmented
- Changed thresholding to Vessel_Thresholded

cjhatch committed:
updated VesselsImage.py to save as a tif

Updated DeviceManager.py to allow for running downstream analysis on thresholded images

Updated gui.py to better handle Measure Vessel Diameter on either Threshold Vessel Images or Segment Vessel Weka. Also set default color to none. Fix issue where you can not select vessel weka as well

cjhatch committed:
updated DeviceImage.py
- increase text box size for selecting ROI

cjhatch committed:
updated DeviceManager.py and VesselsImage.py

- Check image type for Measuring Vessel Diameters and DXF conversion. If it is not thresholded, it prints a warning and skips that image

cjhatch committed:
updated gui.py

- automitically selects tumor weka classifier if tumor classifier is selected
- update GUI to have verbose logging turned on

cjhatch committed:
updated DeviceManager.py

- Fixed issue with confirming each image type, by having operation run before cropping

cjhatch committed:
Update README.md

-Provided cleaner instructions for MacOS install
-Provided clarity on interaction of group cropping with confirm image types

cjhatch committed:
Update README.md

cjhatch committed:
Update README.md

cjhatch committed:
updated PerfusionImage.py

- fixed instructions for setting ROIs for permeability calculation

cjhatch committed:
Update README.md

cjhatch committed:
Update README.md

cjhatch committed:
Update README.md
